sqlobject/src/main/java/org/jdbi/v3/sqlobject/UpdateHandler.java

```diff
@@ -42,45 +42,27 @@
             throw new UnableToCreateSqlObjectException(invalidReturnTypeMessage(method, returnType));
         }
         if (isGetGeneratedKeys) {
-
-            final ResultReturnThing magic = ResultReturnThing.forMethod(sqlObjectType, method);
+            final ResultReturner magic = ResultReturner.forMethod(sqlObjectType, method);
             final GetGeneratedKeys ggk = method.getAnnotation(GetGeneratedKeys.class);
-            final RowMapper<?> mapper;
-            if (DefaultGeneratedKeyMapper.class.equals(ggk.value())) {
-                mapper = new DefaultGeneratedKeyMapper(returnType, ggk.columnName());
-            }
-            else {
-                try {
-                    mapper = ggk.value().newInstance();
-                }
-                catch (Exception e) {
-                    throw new UnableToCreateStatementException("Unable to instantiate row mapper for statement", e, null);
-                }
-            }
-            this.returner = (update, handle) -> {
-                GeneratedKeys<?> o = update.executeAndReturnGeneratedKeys(mapper, ggk.columnName());
-                return magic.result(o, handle);
-            };
+            final RowMapper<?> mapper = ResultReturner.rowMapperFor(ggk, returnType);
+
+            this.returner = magic.bindExecutor(update ->
```

Comment:
The `returner` function and `ResultReturner` probably should be combined somehow